### PR TITLE
Add support for User defined labels on Vault and VaultConfigurer resources

### PR DESCRIPTION
**Pull request by primeroz**
It allows extending the Operator managed labels with custom user defined labels.

Those labels are only applied to the resources and don't propagate in any selector.

### CODE REVIEW

File: docs/annotations-and-labels/README.md [hunk 1]
```diff
@@ -86,3 +86,40 @@ metadata:
 
 These annotations are set *only* on the etcd pods created by the etcd-operator
 
+
+# Labels
+
+The Vault Operator suypport labelling most of the resources it creates using a set of fields in the Vault Specs:
```
Comment (baluchicken) [inline docs/annotations-and-labels/README.md]:
```suggestion
The Vault Operator support labelling most of the resources it creates using a set of fields in the Vault Specs:
```

File: docs/annotations-and-labels/README.md [hunk 2]
```diff
@@ -86,3 +86,40 @@ metadata:
...
+kind: "Vault"
+metadata:
+  name: "vault"
+	spec:
+	  vaultLabels:
+		  example.com/log-format: "json"
+```
+
+These Labels are common to all Vault Statefulset Created resources
+  - Vault Statefulset
```
Comment (baluchicken) [inline docs/annotations-and-labels/README.md]:
```suggestion
    - Vault Statefulset
```

File: docs/annotations-and-labels/README.md (hunk 1 above)
Comment (primeroz) [reply docs/annotations-and-labels/README.md]: thanks, fixed

File: docs/annotations-and-labels/README.md (hunk 2 above)
Comment (primeroz) [reply docs/annotations-and-labels/README.md]: thanks, fixed

**Comment by bonifaido** [comment]
Thanks @primeroz, looks great!

**Pull request merged**